### PR TITLE
feat: release 3.0

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: rockspec/apisix-3.0-0.rockspec [hunk 1]
```diff
@@ -0,0 +1,103 @@
+--
```
Comment (spacewander) [inline rockspec/apisix-3.0-0.rockspec:1]: The rockspec should contain patch version, see https://github.com/apache/apisix/blob/master/rockspec/apisix-2.15.0-0.rockspec

Comment (tzssangglass) [reply rockspec/apisix-3.0-0.rockspec:1]: done

**Review by spacewander — COMMENTED** [review]
We should rename rockspec/apisix-3.0-0.rockspec to add a patch version

**Comment by spacewander** [comment]
> > We should rename rockspec/apisix-3.0-0.rockspec to add a patch version
> 
> you mean rename as `apisix-3.0.0-0.rockspec`?

Yes

**Comment by tzssangglass** [comment]
> Yes

done

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**